tests/unit/components/power-select/before-options-test.js

```diff
@@ -0,0 +1,28 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import Ember from 'ember';
+
+const { Controller, set } = Ember;
+
+moduleForComponent(
+  'power-select/before-options',
+  'Unit | Component | power select before options',
+  {
+    unit: true
+  }
+);
+
```

Comment:
An integration test will partially boot the app (albeit not fully, like an acceptance test). It is also expected to render the component, which will have access to (and render) all subcomponents/services/etc.

The intention of an integration test it so check how the component is working with connected parts (while not fully starting the app) and how it interacts with the user/other layers.

The purpose of a component unit test would be to simply execute the component code to test, for example, some complex function within the component.